src/components/hmi_message_handler/src/mqueue_adapter.cc

```diff
@@ -70,8 +72,13 @@ class ReceiverThreadDelegate : public threads::ThreadDelegate {
     }
   }
 
+  void exitThreadMain() OVERRIDE {
+    shutdown_requested_ = true;
```

Comment:
@pestOO According to [Advanced UNIX Programming](https://books.google.bg/books?id=vlM72G-1VDkC&dq=mq_receive+poll&hl=bg&source=gbs_navlinks_s) the `poll` will not work - [There's no way to tie a message-waiting system call directly to a file descriptor](https://books.google.bg/books?id=vlM72G-1VDkC&pg=PT421&lpg=PT421&dq=mq_receive+poll&source=bl&ots=kJi7iZk4Hs&sig=PuovRw0DN9Cn_o5N5ukFzXz6O_E&hl=bg&sa=X&ved=0ahUKEwj0pMn97JPPAhUISJoKHaEaDbAQ6AEIWzAH#v=onepage&q=mq_receive%20poll&f=false) So the second solution using `O_NONBLOCK` should be used.